src/box/sql/expr.c

```diff
@@ -113,7 +113,7 @@ sql_expr_type(struct Expr *pExpr)
 		assert(pExpr->pRight != NULL && pExpr->pLeft != NULL);
 		enum field_type lhs_type = sql_expr_type(pExpr->pLeft);
 		enum field_type rhs_type = sql_expr_type(pExpr->pRight);
-		return sql_type_result(rhs_type, lhs_type);
+		return sql_type_result(lhs_type, rhs_type);
```

Comment:
`sql: fix order of arguments for sql_type_result` - what kind of weird warning is that? If a function's argument order doesn't matter, what is wrong with writing them in any order?

Can it just be ignored? Same with the `sql_xfree(NULL)` commit.

Reply:
It is possible, but it will appear every time we use `sql_xfree(NULL)` or `sql_type_result(rhs, lhs)`, which is rather inconvenient. At the moment we have two static analyzers that have identified both problems in each case. For `sql_xfree()` the problem sounds like `Null point dereference`, which is unnerving to unprepared observers. I think it is easier to fix the problems.